### PR TITLE
Update PackageValidationBaselineVersion

### PR DESCRIPTION
**Pull request by surayya-MS**
Part of #9914 "Create and merge PR including public API baseline package version change"

### CODE REVIEW

**Comment by rainersigwald** [comment]
This looks syntactically good but we can't merge it until that package is pushed to NuGet.org and the release is done.

File: src/Build/Microsoft.Build.csproj [hunk 1]
```diff
@@ -27,6 +27,11 @@
     <NoWarn>$(NoWarn);NU5104</NoWarn>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <ApiCompatPreserveUnnecessarySuppressions>true</ApiCompatPreserveUnnecessarySuppressions>
```
Comment (maridematte) [inline src/Build/Microsoft.Build.csproj:31]: These are not ideal, but the check is currently failing in a weird way where it is complaining about the comments that we have on the xml files.

**Pull request closed without merging**